bundle/Resources/public/js/views/plugins/cof-copycontentplugin.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('cof-copycontentplugin', function (Y) {
+    "use strict";
+    /**
+     * Provides the Copy Content plugin.
+     *
+     * @module cof-copycontentplugin
+     */
+    Y.namespace('cof.Plugin');
+
+    /**
+     * Provides plugin for the copy content plugin.
+     *
+     * @namespace cof.Plugin
+     * @class CopyContent
+     * @constructor
+     * @extends eZ.Plugin.CopyContent
+     */
+    Y.cof.Plugin.CopyContent = Y.Base.create('cofCopyContentPlugin', Y.eZ.Plugin.CopyContent, [], {
```

Comment:
I would not call that *polluting* but properly design the software to both enhance extensibility and make it future proof (we don't know in advance how the UDW is used).

At the moment, this kind of constraint is expressed in the `isSelectable` function which is an easy way to write the constraint but can only be used when a content is about to be selected (not before). This is an issue for COF but that's also an issue in PlatformUI itself actually. For instance, in the cases where we need a container, we would be able to search only for Container content in the *Search* method.

But it's true this would be a quite big change that deserves some architecture thoughts. After thinking a bit about, there's maybe a *lighter* way of fixing that. If I understand correctly the whole operation flow, I think the `isSelectable` function (for instance [this one](https://github.com/ezsystems/PlatformUIBundle/blob/master/Resources/public/js/views/services/plugins/ez-copycontentplugin.js#L43) when copying) is not called on the newly created content. So you should add that somewhere and if the new Content does not pass this test you could for instance (I'm not sure how much work this involves, feel free to suggest/implement another behavior) get the editor back on the Content Type selector with a message telling him why this happened.